libraries/protobuf/serialization/CMakeLists.txt

```diff
@@ -1,19 +1,16 @@
-cmake_minimum_required(VERSION 3.1.2)
+cmake_minimum_required(VERSION 3.15)
 project(sensor CXX)
 
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
-
-find_package(Protobuf REQUIRED)
+find_package(Protobuf REQUIRED CONFIG)
 
 protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS sensor.proto)
 protobuf_generate_python(PROTO_PYS sensor.proto)
 
 add_executable(${PROJECT_NAME} main.cc ${PROTO_SRCS} ${PROTO_HDRS})
-target_link_libraries(${PROJECT_NAME} PUBLIC CONAN_PKG::protobuf)
+target_link_libraries(${PROJECT_NAME} PRIVATE protobuf::protobuf)
```

Comment:
This PRIVATE might be a bit problematic for some occasional transitive dependencies via CMakeDeps, maybe not for this particular example, but in general using PUBLIC here helps to avoid some failures